redis/src/cluster_async/mod.rs

```diff
@@ -1313,6 +1322,22 @@ where
         }
     }
 
+    async fn handle_loading_error(
+        core: Core<C>,
+        info: RequestInfo<C>,
+        identifier: ConnectionIdentifier,
+    ) -> OperationResult {
+        let is_primary = core.conn_lock.read().await.is_primary(&identifier);
+
+        if !is_primary {
+            // If the connection is a replica, remove the connection and retry.
+            // The connection will be established again on the next call to refresh slots once the replica is no longer in loading state.
```

Comment:
Does the functionality of periodic calls to refresh slots exists. Or is it pending future changes?

Reply:
it exists.

Reply:
We should merge this PR only after we enable the periodic checks